src/main/java/uk/gov/justice/laa/crime/cfecrime/utils/FullMeansTestOutcomeCalculator.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.laa.crime.cfecrime.utils;
+
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.FullAssessmentResult;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.MagCourtOutcome;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.CaseType;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.MeansTestOutcome;
+import java.util.HashSet;
+import java.util.Set;
+
+@Slf4j
+public class FullMeansTestOutcomeCalculator {
+
+    public static MeansTestOutcome getFullMeansTestOutcome(FullAssessmentResult result, CaseType caseType, MagCourtOutcome magCourtOutcome){
+        log.debug("Get the outcome of the Full Means Test result = {} caseType = {} magCourtOutcome = {}", result, caseType, magCourtOutcome);
+
+        //Fail data
+        Set<CaseType> caseTypesHeardInMagistratesCourt = new HashSet<CaseType>();
+        caseTypesHeardInMagistratesCourt .add(CaseType.COMMITAL);
+        caseTypesHeardInMagistratesCourt .add(CaseType.SUMMARY_ONLY);
+        caseTypesHeardInMagistratesCourt .add(CaseType.EITHER_WAY);
+
+        MeansTestOutcome meansTestOutcome = MeansTestOutcome.ELIGIBLE_WITH_NO_CONTRIBUTION;
+        if (result != null && caseType != null ) {
```

Comment:
I don't think we need to check the nullness and throw our own exception - they shouldn't be null, so a NullPointerException would be better if that ever happens (which would be a programmer error as they are NotNull in the JSON specifications)

Reply:
But does not harm if we do check, does it?

Reply:
IMHO it results in a less-useful error message - we just say something bad happened rather than a NullPointerException

Reply:
now outputs the inputs given, which will give and indication.